File: 2075-decode-the-slanted-ciphertext/2075-decode-the-slanted-ciphertext.ts
function decodeCiphertext(encodedText: string, rows: number): string {
    let l=encodedText.length;
    let m=Math.floor(l/rows);

    let orgText:string = "";

    for(let col=0;col<m;col++){
        for(let j=col;j<l;j+=(m+1)){
            orgText+=encodedText[j];
        }
    }

    let ans:string[] = orgText.split("");

    while(ans.length>0 && ans[ans.length-1]===" "){
        ans.pop();
    }

    return ans.join("");
};

File: 2075-decode-the-slanted-ciphertext/2075-decode-the-slanted-ciphertext.cpp
class Solution {
public:
    string decodeCiphertext(string encodedText, int rows) {
        int len = encodedText.length();
        int columns = len/rows;

        string ans="";

        for(int col=0;col<columns;col++){
            for(int j=col;j<len;j+=(columns+1)){
                ans+=encodedText[j];
            }
        }

        while(!ans.empty() && ans.back()==' '){
            ans.pop_back();
        }

        return ans;
    }
};